src/Backend/Synapse/SynapseImportOptions.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Db\ImportExport\Synapse;
+
+use Keboola\Db\ImportExport\ImportOptions;
+
+class SynapseImportOptions extends ImportOptions
+{
+    public const CREDENTIALS_SAS = 'SAS';
+    public const CREDENTIALS_MANAGED_IDENTITY = 'MANAGED_IDENTITY';
+
+    /** @var string */
+    private $importCredentialsType;
+
+    public function __construct(
+        array $convertEmptyValuesToNull = [],
+        bool $isIncremental = false,
+        bool $useTimestamp = false,
+        int $numberOfIgnoredLines = 0,
+        string $importCredentialsType = self::CREDENTIALS_SAS
```

Comment:
Tohle budeme muset přepracovat. To je pain když je tam tolik argumentů. Asi bych to nechal ale na samostatnou story, ať to co nejdřív vyzkoušíme v KBC?

Reply:
Jo klidně, ale nějak mě to nevadí, sů to prostě options, jediná možnost je asi udělat fluent interface, ale takto je to aspoň immutable, fluent interface je :hankey:

Reply:
Můžeš mít fluent interface a immutable, akorát je to pain :) 

```
$this->options()
  ->withHost($someHost);

pubf withHost($host) {
  return Options::fromExisting($this, ['host' => $host]);
}
```

Reply:
Právě to už je lepší mutable :D